promql_tester/condition.py

```diff
@@ -0,0 +1,110 @@
+    _for_seconds: str
+    _interval_seconds: str
+    _threshold: dict
+
+    def __init__(self, raw_config: dict):
+        """
+        raw_config: dictionary representing parsed yaml. Expected to have
+        `expression`, `for_seconds`, `interval_seconds` keys.
+        Optionally can have `threshold` key, which is expected to contain a dictionary
+        with `upper` and `lower` keys.
+        `expression` is a promql expression
+        `for_seconds` are seconds that the condition needs to be evaluated for
+        `interval_seconds` represent the interval of which the condition will be evaluated
+        `threshold` is the threshold for which the value is expected to be in. Defaults to
+        `upper` = 0 and `lower` = 0, meaning that if the result is 0 the condition is successful.
+        """
+        self._raise_if_error(raw_config)
+        self._expression = raw_config["expression"]
+        self._for_seconds = int(raw_config["for_seconds"])
+        self._interval_seconds = int(raw_config["interval_seconds"])
+        self._uuid = uuid.uuid4()
+        self._threshold = self._get_threshold(raw_config)
+
+    def get_id(self) -> str:
+        """Returns autogenerated condition ID"""
+        return self._uuid
+
+    def get_interval_seconds(self):
+        """Returns interval seconds"""
+        return self._interval_seconds
+
+    def get_for_seconds(self):
+        """Returns for seconds"""
+        return self._for_seconds
+
+    def get_expression(self):
+        """Returns expression"""
+        return self._expression
+
+    def is_within_threshold(self, value):
+        """Returs True if given parameter n falls within threshold, False otherwise"""
+        return self._threshold["lower"] <= value <= self._threshold["upper"]
+
+    def __repr__(self):
+        """String representation for condition object"""
+        return f"ID: {self.get_id()}, " + \
+            f"EXPRESSION: <{self.get_expression()}>, " + \
+            f"FOR_SECONDS: <{self.get_for_seconds()}>, " + \
+            f"INTERVAL_SECONDS: <{self.get_interval_seconds()}>"
+
+    def _raise_if_error(self, raw_config):
+        """Raise ConditionConfigException if raw_config is not valid"""
+        for field in self.REQUIRED_FIELDS:
+            if field not in raw_config:
+                raise ConditionConfigException(f"Field {field} not in config: {raw_config}")
+
+    def _get_threshold(self, raw_config):
+        """Returns threshold, with default values if not defined"""
+        if "threshold" not in raw_config:
+            return self.DEFAULT_THRESHOLD
+        threshold = raw_config["threshold"]
+        if "upper" not in threshold:
+            threshold["upper"] = self.DEFAULT_THRESHOLD["upper"]
+        if "lower" not in threshold:
+            threshold["lower"] = self.DEFAULT_THRESHOLD["lower"]
+        threshold["upper"] = float(threshold["upper"])
+        threshold["lower"] = float(threshold["lower"])
+        return threshold
```

Comment:
isn't it equivalent to ?
```suggestion
        return {**self.DEFAULT_THRESHOLD, **raw_config.get("threshold", {})}
```